packages/runtime/src/next/client.ts

```diff
@@ -699,29 +738,33 @@ export class Makeswift {
       response = responseForRequestedLocale
     }
 
-    if (response.status === 404) {
-      return {
-        document: {
-          id,
-          locale: locale ?? null,
-          data: null,
-        },
-        key,
-        cacheData: CacheData.empty(),
-        meta: {
-          allowLocaleFallback,
-          requestedLocale: locale ?? null,
-        },
+    if (!response.ok) {
```

Comment:
Moved the 404 check inside of the `if (!response.ok)` block to match the general pattern throughout this file.